junit/src/main/java/cucumber/api/junit/CucumberRule.java

```diff
@@ -0,0 +1,139 @@
+import static java.util.Arrays.asList;
+
+public class CucumberRule implements TestRule {
+    private final RuntimeOptions runtimeOptions;
+
+    public CucumberRule() {
+        this(new RuntimeOptionsFactory(CucumberRule.class /* no conf */).create());
+    }
+
+    public CucumberRule(final RuntimeOptions runtimeOptions) {
+        this.runtimeOptions = runtimeOptions;
+    }
+
+    public CucumberRule(final String defaultPackage, final String... args) {
+        final List<String> options = new ArrayList<String>(args == null ? Collections.<String>emptyList() : asList(args));
+        if (defaultPackage != null) {
+            if (options.isEmpty()) {
+                options.add(MultiLoader.CLASSPATH_SCHEME + defaultPackage.replace('.', '/'));
+            }
+            if (!hasOption(options, "--glue")) {
+                options.add("--glue");
+                options.add(defaultPackage);
+            }
+            if (!hasOption(options, "--plugin")) {
+                options.add("--plugin");
+                options.add("pretty");
+            }
+        }
+        this.runtimeOptions = new RuntimeOptions(options);
+    }
+
+    private boolean hasOption(List<String> options, String optName) {
+        boolean hasPlugin = false;
+        for (final String opt : options) {
+            if (opt.equals(optName)) {
+                hasPlugin = true;
+                break;
+            }
+        }
+        return hasPlugin;
+    }
+
+    public CucumberRule(final String[] args) {
+        this(null, args);
+    }
+
+    @Override
+    public Statement apply(final Statement base, final Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+                final ResourceLoader resourceLoader = new MultiLoader(classLoader);
+                final ClassFinder classFinder = new ResourceLoaderClassFinder(resourceLoader, classLoader);
+                final Runtime runtime = new Runtime(resourceLoader, classFinder, classLoader, runtimeOptions);
+                final List<CucumberFeature> cucumberFeatures = runtimeOptions.cucumberFeatures(resourceLoader);
+                final JUnitReporter jUnitReporter = new JUnitReporter(runtimeOptions.reporter(classLoader), runtimeOptions.formatter(classLoader), runtimeOptions.isStrict());
+                try {
+                    for (final CucumberFeature feature : cucumberFeatures) {
+                        new FeatureRunner(feature, runtime, jUnitReporter).run(new RunNotifier());
+                    }
+                    base.evaluate();
+                } finally {
+                    jUnitReporter.done();
+                    jUnitReporter.close();
+                    runtime.printSummary();
+                }
+            }
+        };
+    }
+
+    public static class ForceRunner extends BlockJUnit4ClassRunner {
```

Comment:
Make junit happy without loosing stand runner features. Can be replaced with a more advanced/complete custom runner but harder to maintain probably.